cmd/system-probe/api/module/loader.go

```diff
@@ -122,6 +127,8 @@ func Register(cfg *sysconfigtypes.Config, httpMux *mux.Router, factories []Facto
 	l.cfg = cfg
 	if len(l.modules) == 0 {
 		return errors.New("no module could be loaded")
+	} else if !didAnyModuleSucceed {
+		return errors.New("no required module could be loaded successfully")
```

Comment:
what's the reasoning behind this change?

Reply:
System-probe should fail to init if no module starts. However, if we add the process monitor module, we will lose that behavior as that module is likely going to be started anyways. That's also why I didn't move that change to another PR as it's very coupled with this module.